lambda-cloudformation-custom-resource/src/main/scala/feral/lambda/cloudformation/CloudFormationCustomResource.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.cloudformation
+
+import cats.effect._
+import cats.effect.kernel.Resource
+import cats.syntax.all._
+import feral.lambda.cloudformation.CloudFormationCustomResourceHandler.stackTraceLines
+import feral.lambda.cloudformation.CloudFormationRequestType._
+import feral.lambda.{Context, IOLambda}
+import io.circe._
+import io.circe.syntax._
+import org.http4s.Method.POST
+import org.http4s.client.Client
+import org.http4s.circe._
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.ember.client.EmberClientBuilder
+
+import java.io.{PrintWriter, StringWriter}
+
+trait CloudFormationCustomResource[F[_], Input, Output] {
+  def createResource(event: CloudFormationCustomResourceRequest[Input]): F[HandlerResponse[Output]]
+  def updateResource(event: CloudFormationCustomResourceRequest[Input]): F[HandlerResponse[Output]]
+  def deleteResource(event: CloudFormationCustomResourceRequest[Input]): F[HandlerResponse[Output]]
+}
+
+abstract class CloudFormationCustomResourceHandler[Input : Decoder, Output: Encoder]
+  extends IOLambda[CloudFormationCustomResourceRequest[Input], Unit]
+    with Http4sClientDsl[IO] {
+  type Setup = (Client[IO], CloudFormationCustomResource[IO, Input, Output])
+
+  override final def setup: Resource[IO, Setup] =
+    client.mproduct(handler)
+
+  protected def client: Resource[IO, Client[IO]] =
+    EmberClientBuilder.default[IO].build
+
+  def handler(client: Client[IO]): Resource[IO, CloudFormationCustomResource[IO, Input, Output]]
+
+  override def apply(event: CloudFormationCustomResourceRequest[Input],
+                     context: Context,
+                     setup: Setup): IO[Option[Unit]] =
+    (event.RequestType match {
+      case CreateRequest => setup._2.createResource(event)
+      case UpdateRequest => setup._2.updateResource(event)
+      case DeleteRequest => setup._2.deleteResource(event)
```

Comment:
Should we pass the `Context` object to the handler as well? I think you said you've never used it, but maybe there's something useful in there e.g. `awsRequestId` that someone might want to use for logging?